Dev/parser

jieon-seol opened the pull request:
Parser 1차 구현분입니다
DONE: 외부에 제공하는 기능적인 부분(ParserResult를 채우는 것)은 구현완료
TODO: refactoring과 예외처리부분 정책

jieon-seol committed:
[Parser] add file: Parser.cpp/h, ParserTest.cpp

jieon-seol committed:
[Parser] 1차 구현 & Test

jieon-seol committed:
[Parser] OPERATION::ADD시에, ParserResult의 searchColumn, searchData에 사번정보 복사

jieon-seol committed:
Merge branch 'main' into dev/parser

gee6809 reviewed (approved):
LGTM

sky64422 commented on dynamic/Parser.cpp:
```diff
@@ -0,0 +1,125 @@
+	if (paramStr == "DEL") return OPERATION_TYPE::DEL;
+	if (paramStr == "SCH") return OPERATION_TYPE::SCH;
+	if (paramStr == "MOD") return OPERATION_TYPE::MOD;
+	else return OPERATION_TYPE::MAX;	//abnormal case
```
위 if 문들에서 return 으로 모두 처리돼서 'else' 는 없어도 되지 않을까요??

sky64422 replied:
내부 변경이 없는 Input Parameter 는 const 로 해주는 게 좋아보이는데 어떤가요?

jieon-seol replied:
input이 잘못 들어올 경우(채점 TC에서 그런 경우는 없을거라고 했지만)의 예외처리를 위해서
else일 때 OPERATION::MAX를 리턴하게 했습니다.
이 부분은 리팩토링 진행 중인데, error값인 MAX를 리턴할지, Exception을 던질지 고민 중입니다.

jieon-seol replied:
내부 변경이 없는 input parameter들은 const 적용하도록 하겠습니다
감사합니다

sky64422 replied:
네네 좋습니다

jieon-seol replied:
const 일단 적용했습니다

sky64422 commented on dynamic/Parser.cpp:
```diff
@@ -0,0 +1,125 @@
+	string word;
+
+	try {
+		while (getline(ss, word, ',')) {
```
Searcher 에서도 문자열을 구분자로 split 하는 동작을 하는데,
추후에 이 동작을 메소드로 빼서 공통적으로 사용하면 좋을 듯 하네요.

jieon-seol replied:
넵 전체 merge하고 refactoring할 때 공통적으로 사용할 수 있는부분 따로 빼면 될 거 같습니다

sky64422 commented on dynamic/Parser.cpp:
```diff
@@ -0,0 +1,125 @@
+		}
+
+		//TODO: try-catch 추가 (parsing Error)
+		result.operationType = parseOperationType(words[0]);
```
추후 Refactoring 시에 words 의 index 0~9 를 의미있는 이름 define 으로 설정해도 좋을 거 같습니다.

jieon-seol replied:
오 넵 리팩토링시에 반영해보겠습니다

sky64422 commented on dynamic/Parser.h:
```diff
@@ -0,0 +1,22 @@
+	parse():
+	retrun ParserResult
+	*/
+	struct ParserResult parse(std::string queryStirng);
```
'struct' 는 없어도 되지 않나요?

jieon-seol replied:
앗 왜 들어갔는지 모르겠네요;; 삭제하겠습니다

Bammuri reviewed (approved):
TODO를 주석으로 적어놓으신것, 예외처리 구현 TEST 좋은 방향이라 생각합니다.

sky64422 commented on dynamic/Parser.cpp:
```diff
@@ -0,0 +1,125 @@
+	else return " ";	//abnormal case
+}
+
+std::string Parser::parseData(std::string paramStr, INFO_TYPE type) {
```
parseColumn 과 parseData 는 역할이 valid check 외에는 없어보이는데 맞을까요?
메소드가 없어지거나 이름이 바뀌는 게 맞지 않을까 싶기도 한데, 어떤가요?

jieon-seol replied:
맞습니다. 내부적으로 하는 일은 valid check만 합니다.
(지금 상태보다 valid check의 내용은 추가될 수 있음)

parseOperationType()이나, parseOption1() 등 과 달리
input을 변환해서 주는게 아니라, abnormal case말고는 그대로 리턴해주고 있어서
parseXXX()라는 이름이 부적절한거 같다는 말씀이신거죠?

jieon-seol replied:
checkValidColumn, checkValidData로 이름 변경하도록 하겠습니다

sky64422 replied:
네네 오히려 valid check 관련 이름이 맞지 않을까 싶어서 그렇습니다.
parseData 없이 바로 word[i] 대입해도 되지 않나 싶습니다.
동작상 오류는 없고, 변경점이 커서 Integration 후 refactoring 으로 고민해도 될 듯 합니다.

jieon-seol replied:
리팩토링 시에, regex 통한 searchData, changeData의 valid check를 고민중이라서요...
근데 이게 parser에서 할 일이 맞는지 잘 모르겠어서
고민 중입니다.

jieon-seol replied:
일단 파일 이름을 변경했습니다
나머지는 리팩토링시에 더 고민해서 반영하도록 하겠습니다

sky64422 reviewed (commented):
LGTM

Bammuri commented on dynamic/dynamic.vcxproj:
```diff
@@ -140,12 +140,16 @@
   </ItemDefinitionGroup>
```
해당 파일 tracking에서 제외 부탁드려요~

jieon-seol replied:
넵 반영하겠습니다

Bammuri commented on dynamic/dynamic.vcxproj.filters:
```diff
@@ -21,6 +21,7 @@
     <ClInclude Include="Define.h">
```
해당 파일 tracking에서 제외 부탁드려요

jieon-seol replied:
넵 반영하겠습니다

sky64422 reviewed (approved):
LGTM. 고생하셨습니다.

jieon-seol committed:
[Parser] Review 내용 반영
- const 적용
- parseColumn(), parseData() 이름 변경: validCheckColumnName(), validCheckColumnData()
- TYPE_INFO 이름변경: COLUMN_NUM
